src/lib/zkapp.ts

```diff
@@ -826,6 +833,58 @@ Use the optional \`maxTransactionsWithActions\` argument to increase this number
       contract.account.sequenceState.assertEquals(actionsHash);
       return { state, actionsHash };
     },
+    getActions({
+      fromActionHash,
+      endActionHash,
+    }: {
+      fromActionHash?: Field;
+      endActionHash?: Field;
+    }): A[][] {
+      let actionsForAccount: A[][] = [];
+
+      Circuit.asProver(() => {
```

Comment:
I wrapped the logic into `Circuit.asProver` because I am doing string comparisons, that didnt seem to work. Is this a bad thing to do? The logic itself doesnt need to be a part of the circuit, since its just a bit of ordering and sorting

Reply:
it isn't run during compile, so in compile `actionsForAccount` stays undefined

what you want to do is return `actionsForAccount` from a `Circuit.witness` block

Reply:
hm, to return it from a witness block, we have to make it fixed size. however, `reducer.reduce` already handles dynamic size. so.. seems complicated

Reply:
I thought about this again. It's simple actually! Since reduce handles creating the witnesses with the right structure, this doesn't need to deal with that. So, the asProver approach is close to what we want. In compile, the array should just be made empty. If asProver doesn't work, we can differentiate explicitly between the different snark states managed in JS:
inCompile, inAnalayze - in those the array can be empty
inProver, else case - in those the array should be filled

Reply:
I see; its now clearer how exactly asProver and witness behaves. 
Currently, it should return an empty array if asProver wont run - I tested it with the state rollup example and seems to work